### ISSUE TITLE
Ability to hold media in storage folder

### ISSUE THREAD

**Issue by a-komarev** [spatie/laravel-medialibrary#549]
To allow zero down-time deployment [storage:link](https://laravel.com/docs/5.3/filesystem#the-public-disk) command could be used. But after it medialibrary stop working.

There is an error:

> The storage path is not part of the public path

That issue raised a lot earlier. The problem is kinda trivial:

```php
// LocalUrlGenerator.php
/*
 * Get the path where the whole medialibrary is stored.
 */
protected function getStoragePath() : string
{
    $diskRootPath = $this->config->get('filesystems.disks.'.$this->media->disk.'.root');

    return realpath($diskRootPath);
}
```

`realpath` method converts `./public` path to `./storage/public`. Is there any reason to call `realpath`?

I suppose that's pretty common use case and if `realpath` is really required maybe we can just have a clone of the `LocalUrlGenerator` without `realpath` out of the box? For example: `LocalSymlinkUrlGenerator`. Or have a hack in current one :}

**Comment by AlexVanderbist** [comment]
Hi @a-komarev!

Thanks for your input. I've created a PR (#552) that uses the `url` property on the public disk in the `getUrl` method. This should fix your issue without having to change `getStoragePath`!

Feel free to fetch the branch from the PR and try it out. You'll need to make sure that the `url` property on your media disk is set in `config/filesystem.php`. If you're using the default `public` disk it should already be set as follows:
```php
        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
        ],
```

**Comment by a-komarev** [comment]
@AlexVanderbist Maybe better to write it this way?

```php
        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => public_path('storage'),
            'visibility' => 'public',
        ],
```

Or it wouldn't work?

**Comment by AlexVanderbist** [comment]
@a-komarev The `public_path()` method would return the directory's path (for example `/Users/alex/Webdev/laravel-medialibrary-demo/public/storage`) instead of an url.

The url parameter is set to `env('APP_URL').'/storage'` by default in Laravel anyways: https://github.com/laravel/laravel/blob/master/config/filesystems.php#L54

**Comment by a-komarev** [comment]
Yes, you are right.

**Comment by a-komarev** [comment]
I will be able check your PR changes today evening.

**Comment by a-komarev** [comment]
Fixed in #552
